airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -806,10 +806,15 @@ def download_file(
         """
         self.log.info('Downloading source S3 file from Bucket %s with path %s', bucket_name, key)
 
-        if not self.check_for_key(key, bucket_name):
-            raise AirflowException(f'The source file in Bucket {bucket_name} with path {key} does not exist')
-
-        s3_obj = self.get_key(key, bucket_name)
+        try:
+            s3_obj = self.get_key(key, bucket_name)
+        except ClientError as e:
+            if e.response.get('Error', {}).get('Code', {}) == 404:
```

Comment:
```suggestion
            if e.response.get('Error', {}).get('Code') == 404:
```

Should be fine I think